### PR TITLE
Allow passing parameters to request

### PR DESCRIPTION
**Pull request by Gijsvs**
For https://github.com/channable/realtime/pull/1153 I need the ability to pass pagination parameters to woocommerce, this pull request allows me to do this from the realtime code.

### CODE REVIEW

**Comment by Gijsvs** [comment]
@p-nilsson Could you do a review of this? For some reason I can not find your name under reviewers.

File: wc/api.py [hunk 1]
```diff
@@ -54,51 +53,68 @@ def __get_oauth_url(self, url, method):
...
         auth = None
-        params = {}
+
+        if params is None:
+            # If no params have been passed we set it to be an empty dict
+            params = {}
+        elif isinstance(params, dict):
+            # Clear the consumer key and secret if they where passed as parameters.
+            params.pop("consumer_key", None)
+            params.pop("consumer_secret", None)
```
Comment (p-nilsson) [inline wc/api.py]: Perhaps add to the comment that we clear them because they should only be there `self.is_ssl is True and self.query_string_auth is True` and we will take care of that later

File: wc/api.py [hunk 2]
```diff
@@ -54,51 +53,68 @@ def __get_oauth_url(self, url, method):
...
+                        "consumer_key": self.consumer_key,
+                        "consumer_secret": self.consumer_secret,
+                    }
+                )
         else:
             url = self.__get_oauth_url(url, method)
 
         if data is not None:
-            data = jsonencode(data, ensure_ascii=False).encode('utf-8')
+            data = jsonencode(data, ensure_ascii=False).encode("utf-8")
```
Comment (p-nilsson) [inline wc/api.py:109]: Is this change really needed?

**Review by p-nilsson — APPROVED** [review]
LGTM. I would explain why you are removing the consumer secret and key from the params though. It was a bit confusing at first

File: wc/api.py (hunk 2 above)
Comment (Gijsvs) [reply wc/api.py:109]: No, this was blacks doing.

File: wc/api.py (hunk 1 above)
Comment (Gijsvs) [reply wc/api.py]: I will clarify the comment

Comment (Gijsvs) [reply wc/api.py]: Done

**Comment by Gijsvs** [comment]
Sadly this addition does not take into work in combination with oauth url signatures, it seems like we are also able to pass the parameters via the url so we can just revert this change.
